### PR TITLE
Add services for creating a new relation

### PR DESCRIPTION
**Pull request by bonotake**
This PR is the rest of #3 (the other part was #13) and adds an implementation of the `services` for creating a relation.

### CODE REVIEW

File: relations/controllers.py [hunk 1]
```diff
@@ -147,6 +147,11 @@ def supercede(arxiv_id_str: str,
         raise InternalServerError("The previous relation cannot be found") \
             from nfe
 
+    except DBLookUpError as lue:
+        raise InternalServerError("A failure occured in "
+                                  + "looking up the previous relation") \
```
Comment (erickpeirson) [inline relations/controllers.py]:
```suggestion
                                 "looking up the previous relation") \
```

Shouldn't need ``+`` within parentheses

File: relations/controllers.py [hunk 2]
```diff
@@ -209,5 +214,10 @@ def suppress(arxiv_id_str: str,
...
```
Comment (erickpeirson) [inline relations/controllers.py]:
```suggestion
                                  "looking up the previous relation") \
```

File: relations/services/get.py [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
 
 def from_id(id: RelationID) -> Relation:
     """Retrieve a relation from an ID."""
-    pass  # not implemented yet
+    # query to DB
+    try:
+        rels = db.session.query(RelationDB) \
+            .filter(RelationDB.id == int(id)) \
+            .limit(1) \
+            .all()
```
Comment (erickpeirson) [inline relations/services/get.py:26]: Might be more concise to use [``.first()``](https://docs.sqlalchemy.org/en/13/orm/query.html#sqlalchemy.orm.query.Query.first) here -- should generate the same or similar underlying SQL. Returns ``None`` if there is no result.

File: relations/services/tests.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        self.assertEqual(rel_db.resource_type, rel.resource.resource_type)
+        self.assertEqual(rel_db.resource_id, rel.resource.identifier)
+        self.assertEqual(rel_db.description, rel.description)
+        self.assertEqual(rel_db.added_at, rel.added_at)
+        self.assertEqual(rel_db.creator, rel.creator)
+        self.assertEqual(rel_db.supercedes_or_suppresses,
+                         rel.supercedes_or_suppresses)
+
+
+class TestThingGetter(TestCase):
```
Comment (erickpeirson) [inline relations/services/tests.py]:
```suggestion
class TestRelationGetter(TestCase):
```

File: relations/services/tests.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        self.assertEqual(rel.resource.identifier,
+                         self.data['resource_id'])
+        self.assertEqual(rel.description, self.data['description'])
+        self.assertEqual(rel.creator, self.data['creator'])
+        self.assertEqual(rel.added_at, self.data['added_at'])
+        self.assertEqual(rel.supercedes_or_suppresses,
+                         self.data['supercedes_or_suppresses'])
+
+    def test_get_a_relation_that_doesnt_exist(self) -> None:
+        """When the thing doesn't exist, returns None."""
```
Comment (erickpeirson) [inline relations/services/tests.py]:
```suggestion
        """When the relation doesn't exist, raises NotFoundError."""
```

**Review by erickpeirson — APPROVED** [review]
Looks good; a few minor comments. We can further optimize if needed as the rest of the app comes together

File: relations/services/get.py (hunk 3 above)
Comment (bonotake) [reply relations/services/get.py:26]: I rewrote it with `.one()` in another PR, actually.

Comment (bonotake) [reply relations/services/get.py:26]: So I will leave it as it is. We'll change it in another PR.

**Pull request merged**